test/functional/edit-model.js

```diff
@@ -0,0 +1,75 @@
+import { assert } from 'chai';
+import { addAction } from '../util/database';
+import { patch } from '../util/mock-client';
+import server from '../../.dist/lib/server';
+import ItemStore from '../../.dist/store/item-store';
+import ModelStore from '../../.dist/store/model-store';
+
+describe('Edit model', () => {
+
+    before(() => {
+        return server.start(8080);
+    });
+
+    after(() => {
+        return server.stop();
+    });
+
+    beforeEach(() => {
+        return addAction('CLEAR_ALL_DATA').then(() => {
+            return addAction('NEW_MODEL', {
+                name: 'Resistor',
+                description: 'V = IR',
+                manufacturer: 'Pancakes R\' Us',
+                vendor: 'Mouzer',
+                location: 'Shelf 14',
+                isFaulty: false,
+                faultDescription: '',
+                price: 10.50,
+                count: 20
+            });
+        });
+    });
+
+    it('should edit an item', () => {
+        assert.lengthOf(ItemStore.getItems(), 0);
+        let newModel = {
+            name: 'New name',
+            description: 'New description',
+            manufacturer: 'New manufacturer',
+            vendor: 'New vendor',
+            location: 'New location',
+            isFaulty: true,
+            faultDescription: 'it is not working',
+            price: 3.50
+        };
+        return patch('model', {
+            address: 'm8y7nEtAe'
+        }, newModel).then(response => {
+            newModel.address = 'm8y7nEtAe';
```

Comment:
Don't hardcode this. `newModel.address = model.address` is fine since editing a model does not change its address or count.